### PR TITLE
Handle Github rate limit reached

### PR DESCRIPTION
**Pull request by russmatney**
Handles github rate-limited responses in curl-get-json. We now print a message and exit. Includes some docs calling out the env vars for raising the limit.

We may prefer to nil-pun rather than exit. For now I decided to follow the way we exit in the `license` code, which similarly binds `*err*` to `*out*`, prints a message, and exits with an error code.

- [X] This PR corresponds to: https://github.com/babashka/neil/issues/136

- [X] I have updated the [CHANGELOG.md](https://github.com/babashka/neil/blob/main/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

File: README.template.md [hunk 1]
```diff
@@ -207,6 +207,16 @@ Load it using your preferred Emacs package manager, e.g., for Doom Emacs:
...
 ```
 
+## Github's Rate Limit
+
+Github's API has a 60 hit/hour rate-limit. The workaround for this is creating a
+[personal access
+token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+and setting two env vars:
+
+- `BABASHKA_NEIL_DEV_GITHUB_USER`
```
Comment (borkdude) [inline README.template.md:217]: Should we just change this to `NEIL_GITHUB_USER` and `NEIL_GITHUB_TOKEN` so it's not dev specific anymore and everyone can use it?

Comment (russmatney) [reply README.template.md]: @borkdude These are nicer, good thought! I updated the docs and code to read the new env vars. The old ones are still supported for now, but we could drop that support now or in the future - let me know if you feel strongly either way, or if there are any other details you want to address.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## Unreleased
```
Comment (borkdude) [inline CHANGELOG.md:3]: Can you mention the github issue in the changelog?

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**